### PR TITLE
Make it so requisition can spend some of their money into vouchers for dropships.

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
It just adds a little consumable card that you can swipe over the dropship to get points. 
You can buy OB's, why not get CAS some points ? 

## Why It's Good For The Game
More flexibility & some love for CAS

## Changelog
:cl:
add: Added a dropship points voucher to req under OPERATIONS. 
/:cl:

### CODE REVIEW

**Comment by SkeletalElite** [comment]
These are a little expensive. Would cost 120 points to get a rocket like the keeper or even more for a fatty/napalm compared to extra OB prices it's pretty ridiculous for an explosive signficantly smallter. I would reccomend reducing the price a bit.

File: code/modules/reqs/supplypacks.dm [hunk 1]
```diff
@@ -157,6 +157,13 @@ OPERATIONS
 	contains = list(/obj/structure/ob_ammo/ob_fuel)
 	cost = 10
 	containertype = null
+
+/datum/supply_packs/operations/ob_fuel
```
Comment (Moggram) [inline code/modules/reqs/supplypacks.dm:161]:
```suggestion
/datum/supply_packs/operations/cas_voucher

```

**Comment by lbnesquik** [comment]
> 
> 
> These are a little expensive. Would cost 120 points to get a rocket like the keeper or even more for a fatty/napalm compared to extra OB prices it's pretty ridiculous for an explosive signficantly smallter. I would reccomend reducing the price a bit.
> 
> edit: If it helps you decide, the base generation rate of req points is 2/minutes while dropship points is 18/minute, but conversion probably shouldn't be 1:1 it should be a little ineffecient but 1:2.5 is nowhere near the base 1:9 value. I'd reccomend maybe every req point being worth maybe 5 dropship points to start with.

Well how about doubling the amount of dosh to 200 ?

**Comment by Swagile** [comment]
Problem I see here is that this enhances FOB camping a lot more, especially orbit 1 gaus runs (for reference, CAS GAU fires in 1 second after you see the dot on orbit 1). Hence farming a miner or two and giving points to a good CAS pilot means its nearly impossible to siege outside of sitting in caves 24/7 as xenos and hope marines push out. This is especially bad on maps like LV and Ice Colony where there are few cave spots near the main LZ's and the FOB miners.

**Comment by lbnesquik** [comment]
Maybe. I guess it would be best to leave this open and see if the issue remains after we use the other PR

File: code/game/objects/items/misc.dm [hunk 2]
```diff
@@ -175,3 +175,14 @@
 	desc = "A bulky module meant to replace the normal crystalizer in mining wells, used to compress boxes for easy carrying."
 	icon_state = "mining_drill_compactordisplay"
 	uptype = "upgraded crystalizer module"
+
+/obj/item/dropship_points_voucher
+	name = "dropship fabricator voucher"
+	desc = "A small keycard stamped by a Terra Gov logo. It contains points you can redeem at a dropship fabricator. One use only."
+	var/extra_points = 100
```
Comment (BraveMole) [inline code/game/objects/items/misc.dm:182]: new vars need to be autodoc

File: code/game/objects/items/misc.dm [hunk 3]
```diff
@@ -175,3 +175,14 @@
...
+	icon = 'icons/obj/items/card.dmi'
+	icon_state = "centcom"
+
+/obj/item/dropship_points_voucher/examine(mob/user)
+	..()
+	to_chat(user, "It contains [extra_points] points.")
```
Comment (BraveMole) [inline code/game/objects/items/misc.dm:188]: Missing span class

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/items/misc.dm (hunk 2 above)
Comment (lbnesquik) [reply code/game/objects/items/misc.dm:182]: I don't know what autodoc is

File: code/game/objects/items/misc.dm (hunk 3 above)
Comment (lbnesquik) [reply code/game/objects/items/misc.dm:188]: Also clueless

Comment (TiviPlus) [reply code/game/objects/items/misc.dm:188]: Examine should not have span slasses

File: code/game/objects/items/misc.dm [hunk 4]
```diff
@@ -175,3 +175,14 @@
...
+
+/obj/item/dropship_points_voucher
+	name = "dropship fabricator voucher"
+	desc = "A small keycard stamped by a Terra Gov logo. It contains points you can redeem at a dropship fabricator. One use only."
+	var/extra_points = 100
+	icon = 'icons/obj/items/card.dmi'
+	icon_state = "centcom"
+
+/obj/item/dropship_points_voucher/examine(mob/user)
+	..()
```
Comment (TiviPlus) [inline code/game/objects/items/misc.dm:187]:
```suggestion
/obj/item/dropship_points_voucher/examine(mob/user)
	. = ..()
```

File: code/game/objects/machinery/dropship_part_fabricator.dm [hunk 5]
```diff
@@ -96,3 +96,10 @@
 
 		build_dropship_part(build_type, cost, usr)
 		return
+
+/obj/machinery/dropship_part_fabricator/attackby(obj/item/H, mob/user, params)
+	. = ..()
+	if(istype(H, /obj/item/dropship_points_voucher))
+		to_chat(user, "<span class='notice'>You add 100 dropship points to \the [src].</span>")
+		SSpoints.dropship_points += 100
+		qdel(H)
```
Comment (TiviPlus) [inline code/game/objects/machinery/dropship_part_fabricator.dm:105]:
```suggestion
	if(!istype(H, /obj/item/dropship_points_voucher))
		return
	to_chat(user, "<span class='notice'>You add 100 dropship points to \the [src].</span>")
	SSpoints.dropship_points += 100
	qdel(H)
```

File: code/modules/reqs/supplypacks.dm [hunk 6]
```diff
@@ -157,6 +157,14 @@ OPERATIONS
 	contains = list(/obj/structure/ob_ammo/ob_fuel)
 	cost = 10
 	containertype = null
+
+/datum/supply_packs/operations/cas_voucher
+
+	name = "100 dropship fabricator points"
+	contains = list(/obj/item/dropship_points_voucher)
```
Comment (TiviPlus) [inline code/modules/reqs/supplypacks.dm:164]:
```suggestion
/datum/supply_packs/operations/cas_voucher
	name = "100 dropship fabricator points"
	contains = list(/obj/item/dropship_points_voucher)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by lbnesquik** [comment]
Everything should be good now.

File: code/game/objects/items/misc.dm [hunk 7]
```diff
@@ -138,3 +138,22 @@
...
+	icon_state = "mining_drill_compactordisplay"
+	uptype = "upgraded crystalizer module"
+
+/obj/item/dropship_points_voucher
+	name = "dropship fabricator voucher"
+	desc = "A small keycard stamped by a Terra Gov logo. It contains points you can redeem at a dropship fabricator. One use only."
+	///This is the number of points this thing has to give. 
+	var/extra_points = 100
+	icon = 'icons/obj/items/card.dmi'
+	icon_state = "centcom"
```
Comment (TiviPlus) [inline code/game/objects/items/misc.dm:154]:
```suggestion
	icon = 'icons/obj/items/card.dmi'
	icon_state = "centcom"
	///This is the number of points this thing has to give. 
	var/extra_points = 100
```

**Review by TiviPlus — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**